tests/t236_replay_srcline.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+
+from runtest import TestBase
+
+class TestCase(TestBase):
+    def __init__(self):
+        TestBase.__init__(self, 'abc', """
+# DURATION     TID     FUNCTION [SOURCE]
+   0.916 us [  6745] | __monstartup();
+   0.276 us [  6745] | __cxa_atexit();
+            [  6745] | main() { [tests/s-abc.c:26]
+            [  6745] |   a() { [tests/s-abc.c:11]
+            [  6745] |     b() { [tests/s-abc.c:16]
+            [  6745] |       c() { [tests/s-abc.c:21]
+   1.520 us [  6745] |         getpid();
+   2.953 us [  6745] |       } /* c */ [tests/s-abc.c:21]
+   3.636 us [  6745] |     } /* b */ [tests/s-abc.c:16]
+   4.118 us [  6745] |   } /* a */ [tests/s-abc.c:11]
+   4.588 us [  6745] | } /* main */ [tests/s-abc.c:26]
+""", cflags='-g')
+
+    def build(self, name, cflags='', ldflags=''):
+        if cflags.find('-finstrument-functions') >= 0:
+            return TestBase.TEST_SKIP
+
+        return TestBase.build(self, name, cflags, ldflags)
+
+    def prepare(self):
+        self.subcmd = 'record'
+        self.option = '--srcline'
+        return self.runcmd()
+
+    def setup(self):
+        self.subcmd = 'replay'
+        self.option = '--srcline'
+
+    def sort(self, output):
+        """ This function post-processes output of the test to be compared .
+            It ignores blank and comment (#) lines and remaining functions.  """
+        result = []
+        before_main = True
+        for ln in output.split('\n'):
+            if ln.find(' | main()') > 0:
+                before_main = False
+            if before_main:
+                continue
+            # ignore result of remaining functions which follows a blank line
+            if ln.strip() == '':
+                break
+
+            func = ln.split('|', 1)[-1].split('[')
+
+            if len(func) < 2 :
+                result.append('%s' % (func[0]))
+            else :
+                result.append('%s %s' % (func[-2], func[-1][0:-1].split('/')[-1]))
```

Comment:
What is this for?  Do you want to use a basename for the srcline?  I guess we can simply use task_sort (the default sort function) once we can merge #1160, right?

Reply:
Right. It's because #1160 is in progress and not in master for now.

Reply:
Let's get merge it soon. :)

Reply:
I didn't updated sort function for latest change as your comment.

Reply:
If you are planning to merge this after #1160, Is it ok to delete sort function?